File: 0242-valid-anagram/0242-valid-anagram.cpp
#include <map>
using namespace std;
class Solution {
public:
    // O(n log n) - becuase of the sorting time
    bool isAnagram(string s, string t) {
        sort(s.begin(),s.end());
        sort(t.begin(),t.end());
        return s == t;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, stringOne: str, stringTwo: str) -> bool:
        if len(stringOne) != len(stringTwo):
            return False

        balDict = {}

        for char in stringOne:
            if char in balDict:
                balDict[char] += 1
            else:
                balDict[char] = 1
        
        for char in stringTwo:
            if char in balDict:
                balDict[char] -= 1
            else:
                return False
            
        isAnagram = False
        for freq in balDict.values():
            if freq != 0:
                return False


        return True

        